### PR TITLE
feat(structures): Embed interface

### PR DESCRIPTION
**Pull request by matthew119427**
Redo of https://github.com/libraryofcode/dis.ts/pull/24

### CODE REVIEW

File: structures/Embed.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  proxy_url?: string;
+  text: string;
+}
+
+export interface EmbedProvider {
+  name?: string;
+  url?: string;
+}
+
+export type EmbedType = 'rich' | 'image' | 'video' | 'gifv' | 'article' | 'link';
```
Comment (bsian03) [inline structures/Embed.ts]: Enumerate this

File: structures/Embed.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+export default interface Embed {
```
Comment (bsian03) [inline structures/Embed.ts:34]: This should be at the top

**Review by bsian03 — CHANGES_REQUESTED** [review]

File: structures/Embed.ts (hunk 2 above)
Comment (matthew119427) [reply structures/Embed.ts:34]: Moved via commit 3cb89a2

**Comment by matthew119427** [comment]
no because that's exactly what i told her and she never fucking responded because she's a snake
and she and mckenzie know it by god

the next time i see that bitch it's on fuckin sight, hands will be thrown

**Pull request closed without merging**